automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/api/orch/OrchestrationApiSpec.scala

```diff
@@ -63,15 +70,16 @@ class OrchestrationApiSpec extends FreeSpec with Matchers with ScalaFutures with
 
         Orchestration.billing.removeGoogleRoleFromBillingProjectUser(projectName, user.email, role)(ownerToken)
 
-        val postRoleFailure = Retry.retry(1.second, 10.seconds) {
-          // retry this because removing google roles is not always immediate
-          Try { googleBigQueryDAO.startQuery(userToken.value, GoogleProject(projectName), shakespeareQuery).failed.futureValue }.toOption
-        }.get
+        // The google role might not have been removed the first time we call startQuery() - poll until it has
+        eventually {
+          val postRoleFailure = googleBigQueryDAO.startQuery(userToken.value, GoogleProject(projectName), shakespeareQuery).failed.futureValue
+
+          postRoleFailure shouldBe a[GoogleJsonResponseException]
```

Comment:
If my understanding of the pattern above is correct, then I think the `should`s below this line should be outside of the `eventually`. You want to break out of the `eventually` as soon as you have the type of response you expect. By making the other checks inside, you'll also retry if any of the assertions fail. For example, if Google changes the message to not include the project name, the conditions will never be met and retries will happen unnecessarily.

Reply:
OK, this change makes sense.